integration_test/tracing/tracing_test.go

```diff
@@ -0,0 +1,871 @@
+	config.Set("archival.Enabled", false)
+	config.Set("Reporting.syncer.enabled", false)
+	config.Set("BatchRouter.mainLoopFreq", "1s")
+	config.Set("BatchRouter.uploadFreq", "1s")
+	config.Set("Gateway.webPort", strconv.Itoa(port))
+	config.Set("RUDDER_TMPDIR", os.TempDir())
+	config.Set("recovery.storagePath", path.Join(tmpDir, "/recovery_data.json"))
+	config.Set("recovery.enabled", false)
+	config.Set("Profiler.Enabled", false)
+	config.Set("Gateway.enableSuppressUserFeature", false)
+
+	config.Set("enableStats", true)
+	config.Set("RuntimeStats.enabled", false)
+	config.Set("OpenTelemetry.enabled", true)
+	config.Set("OpenTelemetry.traces.endpoint", zipkinURL)
+	config.Set("OpenTelemetry.traces.samplingRate", 1.0)
+	config.Set("OpenTelemetry.traces.withSyncer", true)
+	config.Set("OpenTelemetry.traces.withZipkin", true)
+	config.Set("OpenTelemetry.metrics.prometheus.enabled", true)
+	config.Set("OpenTelemetry.metrics.prometheus.port", prometheusPort)
+	config.Set("OpenTelemetry.metrics.exportInterval", 10*time.Millisecond)
+
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("panicked: %v", r)
+		}
+	}()
+	r := runner.New(runner.ReleaseInfo{EnterpriseToken: "DUMMY"})
+	c := r.Run(ctx, []string{"rudder-tracing"})
+	if c != 0 {
+		err = fmt.Errorf("rudder-server exited with a non-0 exit code: %d", c)
+	}
+	return
+}
+
+// nolint: unparam
+func requireJobsCount(
+	t *testing.T,
+	db *sql.DB,
+	queue, state string,
+	expectedCount int,
+) {
+	t.Helper()
+
+	require.Eventually(t, func() bool {
+		var jobsCount int
+		require.NoError(t, db.QueryRow(fmt.Sprintf(`SELECT count(*) FROM unionjobsdbmetadata('%s',1) WHERE job_state = '%s' AND parameters->>'traceparent' is not NULL;`, queue, state)).Scan(&jobsCount))
+		t.Logf("%s %sJobCount: %d", queue, state, jobsCount)
+		return jobsCount == expectedCount
+	},
+		20*time.Minute,
+		1*time.Second,
+		fmt.Sprintf("%d %s events should be in %s state", expectedCount, queue, state),
+	)
+}
+
+// nolint: unparam
+func sendEvents(
+	num int,
+	eventType, writeKey,
+	url string,
+) error {
+	for i := 0; i < num; i++ {
+		payload := []byte(fmt.Sprintf(`
+			{
+			  "batch": [
+				{
+				  "userId": %[1]q,
+				  "type": %[2]q,
+				  "context": {
+					"traits": {
+					  "trait1": "new-val"
+					},
+					"ip": "14.5.67.21",
+					"library": {
+					  "name": "http"
+					}
+				  },
+				  "timestamp": "2020-02-02T00:23:09.544Z"
+				}
+			  ]
+			}`,
+			rand.String(10),
+			eventType,
+		))
+		req, err := http.NewRequest(http.MethodPost, url+"/v1/batch", bytes.NewReader(payload))
+		if err != nil {
+			return err
+		}
+		req.SetBasicAuth(writeKey, "password")
+
+		resp, err := (&http.Client{}).Do(req)
+		if err != nil {
+			return err
+		}
+
+		if resp.StatusCode != http.StatusOK {
+			b, _ := io.ReadAll(resp.Body)
+			return fmt.Errorf("failed to send event to rudder server, status code: %d: %s", resp.StatusCode, string(b))
+		}
+		func() { kithttputil.CloseResponse(resp) }()
```

Comment:
Do we need to wrap it this way? :thinking:

Reply:
For linter. `response body must be closed (bodyclose)`

Reply:
Darn linter wants us to wrap it? I'd rather not wrap and add a `nolint` tbh. Up to you :+1: